kohese-portal/server/kdb-repo.ts

```diff
@@ -218,45 +218,58 @@ export class KDBRepo {
-            return index.writeTree().then(function (treeId) {
-              var parentCommits = [];
-              if (commit) {
-                parentCommits.push(commit);
-              }
-              return repo.getStatusExt().then(function (statuses) {
-                var filePaths = [];
-                for (var j = 0; j < statuses.length; j++) {
-                  var status = statuses[j];
-                  if (status.inIndex()) {
-                    filePaths.push(status.path());
-                  }
+    async function commitARepo (repo, repositoryId, iIndex) {
+      promises.push(repo.getHeadCommit().then(function (commit) {
+        return repo.refreshIndex().then(function (index) {
+          return index.writeTree().then(function (treeId) {
+            let parentCommits = [];
+            if (commit) {
+              parentCommits.push(commit);
+            }
+            return repo.getStatusExt().then(function (statuses) {
+              let filePaths = [];
+              for (let j = 0; j < statuses.length; j++) {
+                let status = statuses[j];
+                if (status.inIndex()) {
+                  filePaths.push(status.path());
                 }
-
+              }
+              if (filePaths.length > 0) {
                 return repo.createCommit('HEAD', signature, signature, message,
-                    treeId, parentCommits).then(function (commitId) {
-                  commitIdMap[repositoryIds[iIndex]] = {
-                    commitId: commitId,
-                    filesCommitted: filePaths
-                  };
-                  return repo.getCommit(commitId).then(function (c) {
-                    return Promise.resolve(true);
+                  treeId, parentCommits).then(function (commitId) {
+                    commitIdMap[repositoryId] = {
+                      commitId: commitId,
+                      filesCommitted: filePaths
+                    };
+                    return repo.getCommit(commitId).then(function (c) {
+                      return Promise.resolve(true);
+                    }).catch(function (err) {
+                      console.log(err.stack);
+                    });
+                  }).catch(function (err) {
+                    console.log(err.stack);
                   });
-                });
-              });
+              } else {
+                return Promise.resolve(true);
+              }
+            }).catch(function (err) {
+              console.log(err.stack);
             });
           }).catch(function (err) {
             console.log(err.stack);
           });
-        }));
-      })(i);
+        }).catch(function (err) {
+          console.log(err.stack);
+        });
+      }));
+    }
+    for (let repositoryId in repoList) {
+      var repo = repoList[repositoryId];
+      await commitARepo (repo, repositoryId, repositoryId)
     }
 
     return Promise.all(promises).then(function () {
```

Comment:
Since we are now doing the await for each repo, probably don't need this Promise.all.

Reply:
Open issue in Kohese to fix this "Commit Function in KDBRepo should not return a promise" (40137f60-9d56-11eb-b934-f70fd771e96e)